test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -133,8 +133,10 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
 			nil,
 		}
 		for i, w := range workloads {
-			w.Spec.Admission = admissions[i]
-			gomega.Expect(k8sClient.Update(ctx, w)).To(gomega.Succeed())
+			var newWL kueue.QueuedWorkload
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(w), &newWL)).To(gomega.Succeed())
```

Comment:
this might still fail. Better have a retry loop using `gomega.Eventually`. Since the code will be used in a few places, put it in a helper function.